src/vm/jitinterface.cpp

```diff
@@ -11623,13 +11623,21 @@ void CEEJitInfo::allocMem (
 
     S_SIZE_T totalSize = S_SIZE_T(codeSize);
 
+    bool roData16bytesAligned = ((flag & CORJIT_ALLOCMEM_FLG_RODATA_16BYTE_ALIGN)!= 0);
     if (roDataSize > 0)
     {
-        totalSize.AlignUp(sizeof(void *));
+        if (roData16bytesAligned)
+        {
+            totalSize.AlignUp(16);
+        }
+        else
+        {
+            totalSize.AlignUp(sizeof(void *));
+        }
         totalSize += roDataSize;
 
 #ifndef _WIN64
-        if (roDataSize >= 8)
+        if ((roDataSize >= 8) && !roData16bytesAligned)
```

Comment:
Now that I have taken a second look: bad things may happen if somebody uses this flag on !AMD64 because of there won't be padding to use for the actual pointer alignment later in this method. 

We need to add the padding whenever `roDataSizeAlignment > CODE_SIZE_ALIGN`. Or we can make it simple and just force the code to be 16 byte aligned by setting `CORJIT_ALLOCMEM_FLG_16BYTE_ALIGN` flag here if `roDataSizeAlignment > CODE_SIZE_ALIGN`.